sdk/corev2/azure-core/src/main/java/com/azure/v2/core/http/pipeline/BearerTokenAuthenticationPolicy.java

```diff
@@ -137,11 +136,7 @@ public Response<BinaryData> process(HttpRequest httpRequest, HttpPipelineNextPol
         if (httpResponse.getStatusCode() == 401 && authHeader != null) {
             if (authorizeRequestOnChallenge(httpRequest, httpResponse)) {
                 // body needs to be closed or read to the end to release the connection
-                try {
-                    httpResponse.close();
-                } catch (IOException e) {
-                    throw LOGGER.logThrowableAsError(new RuntimeException(e));
```

Comment:
this is an example where we don't even follow `UncheckedIOException` contract and users will need to catch all runtime exceptions if they want to log or handle them in some way.